onnx/test/schema_test.py

```diff
@@ -257,6 +257,72 @@ def test_init_with_default_value(self):
         self.assertEqual("attr1", attribute.name)
         self.assertEqual("attr1 description", attribute.description)
 
+class TestOpSchemaRegister(unittest.TestCase):
+    def test_register(self):
+        input = """
+           agraph (float[N, 128] X, int32 Y) => (float[N] Z)
+           {
+              Z = CustomOp<attr1=[1,2]>(X, Y)
+           }
+           """
+        model = onnx.parser.parse_graph(input)
+        op_schema = defs.OpSchema(
+            "CustomOp",
+            "",
+            1,
+            inputs=[
+                defs.OpSchema.FormalParameter("input1", "T"),
+                defs.OpSchema.FormalParameter("input2", "int32"),
+            ],
+            outputs=[
+                defs.OpSchema.FormalParameter("output1", "T"),
+            ],
+            type_constraints=[("T", ["tensor(float)"], "")],
+            attributes=[
+                defs.OpSchema.Attribute(
+                    "attr1", defs.OpSchema.AttrType.INTS, "attr1 description"
+                )
+            ],
+        )
+        self.assertFalse(onnx.defs.has(op_schema.name))
+        with self.assertRaises(onnx.checker.ValidationError):
+            onnx.checker.check_graph(model)
+        onnx.defs.register_schema(op_schema)
+        onnx.checker.check_graph(model)
+        
+        # cleanup
+        onnx.defs.deregister_schema(op_schema.name, op_schema.since_version, op_schema.domain)
+
```

Comment:
Move the cleanup part to the `cleanUp` method so that it is always run even when any lines above fails.